worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageMappingCallable.java

```diff
@@ -159,26 +157,42 @@ private boolean filterByYear(IDomainResource patient) {
      * @return patientId if present, null otherwise
      */
     private Identifiers extractPatientId(IDomainResource patient) {
+        List<PatientIdentifier> ids = IdentifierUtils.getIdentifiers(patient);
         // Get patient beneficiary id
         // if not found eobs cannot be looked up so do not return a meaningful list
-        String beneId = IdentifierUtils.getBeneId(patient);
-        if (beneId == null || beneId.isEmpty()) {
+        PatientIdentifier beneIdObj = IdentifierUtils.getBeneId(ids);
+        String beneId = null;
+        if (beneIdObj == null) {
             missingBeneId += 1;
             return null;
+        } else {
+            beneId = beneIdObj.getValue();
+            if (beneId == null || beneId.isEmpty()) {
+                missingBeneId += 1;
+                return null;
+            }
         }
 
         // Get current mbi if present or else log not present
-        String currentMbi = IdentifierUtils.getCurrentMbi(patient);
+        PatientIdentifier currentMbi = IdentifierUtils.getCurrentMbi(ids);
+        String curValue = null;
         if (currentMbi == null) {
             missingCurrentMbi += 1;
+        } else {
+            curValue = currentMbi.getValue();
         }
 
-        LinkedHashSet<String> historicMbis = IdentifierUtils.getHistoricMbi(patient);
+        LinkedHashSet<PatientIdentifier> historicMbis = IdentifierUtils.getHistoricMbi(ids);
+
```

Comment:
Line 187 is dubious, why are we deriving logic from a null reference to a collection? Collections references should never be null and instead point to empty collections.